src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLoggerScope.cs

```diff
@@ -0,0 +1,84 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+#if NET451
+using System.Runtime.Remoting;
+using System.Runtime.Remoting.Messaging;
+#else
+using System.Threading;
+#endif
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs
+{
+    public class EventFlowLoggerScope
+    {
+        internal EventFlowLoggerScope(object state)
+        {
+            State = state;
+        }
+
+        public Object State { get; private set; }
+        public EventFlowLoggerScope Parent { get; private set; }
+
+#if NET451
+        private static readonly string FieldKey = $"{typeof(EventFlowLoggerScope).FullName}_{AppDomain.CurrentDomain.Id}";
+        public static EventFlowLoggerScope Current
+        {
+            get
+            {
+                ObjectHandle handle = (ObjectHandle)CallContext.LogicalGetData(FieldKey);
```

Comment:
Can you put in a comment explaining why we are using ObjectHandle here and not just storing the the EventFlowLoggerScope directly?